php7/README

```diff
@@ -0,0 +1,22 @@
+================================================================================
+  PHP 7.3 yacc grammar
+================================================================================
+
+This is an adaption of PHP's 7.3 Yacc grammar for lrpar. There are few things
+that are currently not supported. First, the dangling else problem in PHP is
+resolved by adding the non-existent `T_NOELSE` token, which takes precedence
+over `else`. But since this token can't exist, shifting `else` is chosen in
+favour of a reduction. Since lrpar requires all tokens to be defined in the
+lexer, the `T_NOELSE` token has been removed. However, since lrpar by default
+resolves conflicts in favour of shifts, we get the same outcome, though will
+produce a warning.
+
+Second, PHP lexes strings differently, depending on their contents. For
+example, a string containing variables, like `"hello ${name}"`, will be parsed
+as `T_STRING_VARNAME`. In this grammar, currently all strings are lexed into
+`T_CONSTANT_ENCAPSED_STRING`. Also currently not supported are HEREDOCs and
+inlined HTML.
+
+Third, definitions for tokens that are thrown away by the lexer (e.g.
+T_COMMENT) or do not need to be handled by the parser (e.g. T_OPEN_TAG) have
+been commented out.
```

Comment:
What are the consequences of this?

Reply:
This shouldn't have any consequences as the stuff is never used by the grammar. `T_OPEN_TAG` is `<?php`, so we either throw this away in the lexer, if the input file has those tags, or we don't need it at all, if our input files don't contain these tags.

Reply:
OK, so this is just a "difference" but not really something that's "not supported"?